pkg/handler/aws/uploadProfilePicture/main.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/CollActionteam/collaction_backend/auth"
+	"github.com/CollActionteam/collaction_backend/internal/uploads"
+	awsRepository "github.com/CollActionteam/collaction_backend/pkg/repository/aws"
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-sdk-go/aws/session"
+)
+
+func handler(ctx context.Context, req events.APIGatewayV2HTTPRequest) (res events.APIGatewayProxyResponse, err error) {
+
+	// This is to recover from a "invalid memory address or nil pointer dereference: errorString" runtime error
+	// when the endpoint is called without valid auth token
+	defer func() {
+		if r := recover(); r != nil {
```

Comment:
can you find the source of panic and do some checks to prevent panic?

Reply:
as we don't know if is there any possibility of panic, it's best to keep this code, but log the panic reason to fix it in the future

Reply:
The function can't be called without a valid user, because the API Gateway enforces this through the Authorizer that is used on this handler (see `/template.yaml`).